src/map/script.cpp

```diff
@@ -21030,17 +21031,17 @@ BUILDIN_FUNC(getrandgroupitem) {
 	return SCRIPT_CMD_SUCCESS;
 }
 
-/* getgroupitem <group_id>{,<char_id>};
+/* getgroupitem <group_id>{,<identify>{,<char_id>}};
  * Gives item(s) to the attached player based on item group contents
  */
 BUILDIN_FUNC(getgroupitem) {
 	TBL_PC *sd;
 	int group_id = script_getnum(st,2);
 	
-	if (!script_charid2sd(3,sd))
+	if (!script_charid2sd(4,sd))
 		return SCRIPT_CMD_SUCCESS;
 	
-	if (itemdb_pc_get_itemgroup(group_id,sd)) {
+	if (itemdb_pc_get_itemgroup(group_id, (script_hasdata(st, 3) ? script_getnum(st, 3) != 0 : 0), sd)) {
```

Comment:
Doesn't this give a bool warning again?
should be `: false` instead of `: 0`

Reply:
ok

Reply:
he will have to cast anyway. since the other result of the ternary operator is script_getnum(st, 3).
Well that a bit what was done with != 0.
if (itemdb_pc_get_itemgroup(group_id,(bool) (script_hasdata(st, 3) ? script_getnum(st, 3) : 0), sd))